### PR TITLE
nanodbc: init at 2.13.0

### PR DESCRIPTION
**Pull request by bzizou**
###### Motivation for this change
Package request #130134

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by c0bw3b** [comment]
Friendly ping to @NixOS/darwin-maintainers to maybe try and build this on Darwin. :)
It's not obvious what the problem can be with ofborg failures (no framework mentioned).

**Comment by prusnak** [comment]
Try this change:

```
# backport the change in https://github.com/nanodbc/nanodbc/issues/274
postUnpack = ''
  mv VERSION VERSION.txt
'';
```

File: pkgs/development/libraries/nanodbc/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ unixODBC ];
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DBUILD_STATIC_LIBS=ON"
+  ];
```
Comment (c0bw3b) [inline pkgs/development/libraries/nanodbc/default.nix:21]: We'd want to build the shared lib only by default.
And the static lib when requested.

```suggestion
  cmakeFlags = 
    lib.optionals (!stdenv.hostPlatform.isStatic) [ "-DBUILD_SHARED_LIBS=ON" ]
    ++ lib.optionals (stdenv.hostPlatform.isStatic) [ "-DBUILD_STATIC_LIBS=ON" ];
```

File: pkgs/development/libraries/nanodbc/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+  meta = with lib; {
+    homepage = "https://github.com/nanodbc/nanodbc";
```
Comment (c0bw3b) [inline pkgs/development/libraries/nanodbc/default.nix:24]: You can add `meta.changelog` to help future reviews when updating the package
```suggestion
    homepage = "https://github.com/nanodbc/nanodbc";
    changelog = "https://github.com/nanodbc/nanodbc/raw/v${version}/CHANGELOG.md";
```

**Comment by smaret** [comment]
On darwin-x86, `nix build .#nanodbc` fails with the following error:

```
error: builder for '/nix/store/5mi5mv8c0z3l90kh4sxnyjqy35x7w8gn-nanodbc-2.13.0.drv' failed with exit code 2;
       last 10 log lines:
       > _Tp* __cxx_atomic_fetch_sub(__cxx_atomic_base_impl<_Tp*> volatile* __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
       >                                                                         ^
       > /nix/store/64nxa8d4zwvfycksdf3ahj4rxr2a08gj-libcxx-11.1.0-dev/include/c++/v1/atomic:1073:65: error: unknown type name 'ptrdiff_t'
       > _Tp* __cxx_atomic_fetch_sub(__cxx_atomic_base_impl<_Tp*> * __a, ptrdiff_t __delta, memory_order __order) _NOEXCEPT {
       >                                                                 ^
       > fatal error: too many errors emitted, stopping now [-ferror-limit=]
       > 20 errors generated.
       > make[2]: *** [CMakeFiles/nanodbc.dir/build.make:76: CMakeFiles/nanodbc.dir/nanodbc/nanodbc.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:151: CMakeFiles/nanodbc.dir/all] Error 2
       > make: *** [Makefile:146: all] Error 2
       For full logs, run 'nix log /nix/store/5mi5mv8c0z3l90kh4sxnyjqy35x7w8gn-nanodbc-2.13.0.drv'.
```

File: pkgs/development/libraries/nanodbc/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/nanodbc/default.nix:21]:
```suggestion
  cmakeFlags = if (stdenv.hostPlatform.isStatic) then
    [ "-DBUILD_STATIC_LIBS=ON" ]
  else
   [ "-DBUILD_SHARED_LIBS=ON" ];
```

**Comment by bzizou** [comment]
Thank you @prusnak ! This seems OK now!

File: pkgs/development/libraries/nanodbc/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  cmakeFlags = if (stdenv.hostPlatform.isStatic) then
+    [ "-DBUILD_STATIC_LIBS=ON" ]
+  else
+    [ "-DBUILD_SHARED_LIBS=ON" ];
+
+  # fix compilation on macOS
+  # https://github.com/nanodbc/nanodbc/issues/274
+  # remove after the next version update
+  postUnpack = if stdenv.isDarwin then ''
+    mv $sourceRoot/VERSION $sourceRoot/VERSION.txt
```
Comment (smaret) [inline pkgs/development/libraries/nanodbc/default.nix:27]: You need to add this as well:
```
    substituteInPlace $sourceRoot/CMakeLists.txt \
       --replace 'file(STRINGS VERSION' 'file(STRINGS VERSION.txt'
```

**Review by smaret — CHANGES_REQUESTED** [review]

File: pkgs/development/libraries/nanodbc/default.nix (hunk 3 above)
Comment (bzizou) [reply pkgs/development/libraries/nanodbc/default.nix:27]: done

**Comment by bzizou** [comment]
So, this can be merged now, isn't it?

File: pkgs/development/libraries/nanodbc/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  cmakeFlags = if (stdenv.hostPlatform.isStatic) then
+    [ "-DBUILD_STATIC_LIBS=ON" ]
+  else
+    [ "-DBUILD_SHARED_LIBS=ON" ];
+
+  # fix compilation on macOS
+  # https://github.com/nanodbc/nanodbc/issues/274
+  # remove after the next version update
+  postUnpack = if stdenv.isDarwin then ''
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nanodbc/default.nix:26]: Why is this not using lib.optionalString?

File: pkgs/development/libraries/nanodbc/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    repo = "nanodbc";
+    rev = "v${version}";
+    sha256 = "1q80p7yv9mcl4hyvnvcjdr70y8nc940ypf368lp97vpqn5yckkgm";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ unixODBC ];
+
+  cmakeFlags = if (stdenv.hostPlatform.isStatic) then
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/nanodbc/default.nix:18]: The `()` are optional

**Pull request merged**